span.go

```diff
@@ -119,9 +119,13 @@ func (s *spanImpl) FinishWithOptions(opts opentracing.FinishOptions) {
 	if s.tracer.Options.DebugAssertUseAfterFinish {
 		// This makes it much more likely to catch a panic on any subsequent
 		// operation since s.tracer is accessed on every call to `Lock`.
+		pool := s.tracer.spanPool
```

Comment:
why is retrieving spanPool variable here makes a difference? The code looks functionally identical to what was before.